### DISCUSSION TITLE
Add Alchemy/Infura Providers with WagmiAdapter

### DISCUSSION THREAD

**Discussion by Idan-Shperling** [reown-com/appkit#3170] [Q&A] [answered]
Do you know how I can add an Infura/Alchemy provider?
Will I have rate limit issues?

**Comment by italoHonoratoSA** [comment]
See if this documentation helps with your problem:
https://docs.reown.com/appkit/javascript/core/custom-networks

**Reply by Idan-Shperling** [reply]
Thanks. I saw that, but I have many networks, like polygon, base, avalanche, etc. Do I need to create each one as a custom network just to use our RPC URL? Is there a better way to do so?

**Reply by vittominacori** [reply, marked as answer]
Note that, also if you define your own rpc, the default rpc will be (always) used for supported network as discussed in that issue https://github.com/reown-com/appkit/issues/3187 and defined here https://github.com/reown-com/appkit/blob/main/packages/appkit-utils/src/CaipNetworkUtil.ts#L115-L117

**Reply by Idan-Shperling** [reply]
ok got it. thanks
